packages/eas-cli/src/build/android/build.ts

```diff
@@ -119,8 +122,17 @@ async function ensureAndroidCredentialsAsync(
     }),
     {
       app: {
+        account: nullthrows(
+          findAccountByName(ctx.commandCtx.user.accounts, ctx.commandCtx.accountName),
+          `You do not have access to account: ${ctx.commandCtx.accountName}`
+        ),
         projectName: ctx.commandCtx.projectName,
-        accountName: ctx.commandCtx.accountName,
+        androidApplicationIdentifier: nullthrows(
```

Comment:
This value from app.json is not correct for generic projects. Actually, we even print a warning if someone is building a generic project and they have `android.package` defined in their app.json (see https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/project/android/applicationId.ts#L35). For generic, we read the application id from native code. 

Take a look at line 70 in this file and use the value returned by the `getOrConfigureApplicationIdAsync` function.

Reply:
good catch, thanks 🙏